set rust crate lto to thin to not use fatlto for gkrust which fails sometimes

additionally, disable lto on riscv64 altogether

--- a/config/makefiles/rust.mk
+++ b/config/makefiles/rust.mk
@@ -56,7 +56,7 @@ cargo_build_flags += $(MOZ_CARGO_BUILD_S
 cargo_rustc_flags = $(CARGO_RUSTCFLAGS)
 # Enable link-time optimization for release builds.
 ifdef RUST_LIBRARY_LTO
-cargo_rustc_flags += -Clto
+cargo_rustc_flags += -Clto=thin
 endif
 
 ifdef CARGO_INCREMENTAL
